### PR TITLE
Add optparser.rbi to stdlib

### LINKED ISSUE #1263: OptionParser isn't included in the stdlib
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0A%0Arequire%20'optparse'%0A%0Aoptions%20%3D%20%7B%7D%0AOptionParser.new%20do%20%7Copts%7C%0A%20%20opts.banner%20%3D%20%22Usage%3A%20example.rb%20%5Boptions%5D%22%0A%0A%20%20opts.on(%22-v%22%2C%20%22--%5Bno-%5Dverbose%22%2C%20%22Run%20verbosely%22)%20do%20%7Cv%7C%0A%20%20%20%20options%5B%3Averbose%5D%20%3D%20v%0A%20%20end%0Aend.parse!%0A%0Ap%20options%0Ap%20ARGV)

```ruby
# typed: true

require 'optparse'

options = {}
OptionParser.new do |opts|
  opts.banner = "Usage: example.rb [options]"

  opts.on("-v", "--[no-]verbose", "Run verbosely") do |v|
    options[:verbose] = v
  end
end.parse!

p options
p ARGV
```

This code comes directly from the [Ruby documentation](https://ruby-doc.org/stdlib-2.5.1/libdoc/optparse/rdoc/OptionParser.html).

#### Observed output

```
editor.rb:6: Unable to resolve constant OptionParser https://srb.help/5002
     6 |OptionParser.new do |opts|
        ^^^^^^^^^^^^
    https://github.com/sorbet/sorbet/tree/beecfbcc014c33282ea65f03a61fbb74535a5435/rbi/stdlib/net.rbi#L1606: Did you mean: Net::HTTP::Options?
    1606 |class Net::HTTP::Options < Net::HTTPRequest
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    https://github.com/sorbet/sorbet/tree/beecfbcc014c33282ea65f03a61fbb74535a5435/rbi/core/rational.rbi#L3: Did you mean: Rational?
     3 |class Rational < Numeric
        ^^^^^^^^^^^^^^^^^^^^^^^^
Errors: 1
```

#### Expected behavior

I'd expect OptionParser to be defined in `rbi/stdlib` (or `rbi/core`?)

See the Ruby Documentation: https://ruby-doc.org/stdlib-2.5.1/libdoc/optparse/rdoc/OptionParser.html

### PR DESCRIPTION
**Pull request by connorshea**
This fixes #1263. Based on the `optparse.rb` file in Ruby itself: https://github.com/ruby/ruby/blob/v2_6_3/lib/optparse.rb

I used [Sord](https://github.com/AaronC81/sord) to create this skeleton, and it seems to have worked relatively well even without any YARD docs (though, everything is untyped obviously). I replaced a few `returns` with `void` for some `initialize` methods, but other than that I didn't mess with it much. Another concern is that Sord replaced the attr_readers/writers with the methods that Ruby actually generates, even though Sorbet can handle that itself. I'm not sure if that's a huge problem, but I figured I'd mention it.

I'm not sure if `private` methods should be excluded from this `rbi`?

My only other question is how I should handle the last line of `optparser.rb`, where the `OptParser` is made an alias of `OptionParser`: https://github.com/ruby/ruby/blob/v2_6_3/lib/optparse.rb#L2191

### Motivation

I wanted to have OptionParser not throw Sorbet errors when it's used.

### Test plan
I didn't see any tests for other stdlib libraries, so I didn't include any here.

### CODE REVIEW

**Comment by connorshea** [comment]
Oh, it also added each of `ParseError`s methods to the child classes explicitly, I can clean that up if we want.

**Comment by connorshea** [comment]
Ran the expectation generator and now the tests all pass locally 🎉 

Only ones I'm unsure of are the MissingArgument suggestions for MissingConstant - but I assume that's probably fine.

File: rbi/stdlib/optparser.rbi [hunk 1]
```diff
@@ -0,0 +1,778 @@
+# typed: __STDLIB_INTERNAL
+class OptionParser
+  sig { params(to: T.untyped, name: T.untyped).returns(T.untyped) }
```
Comment (DarkDimius) [inline rbi/stdlib/optparser.rbi]: Our current `rbi`'s use a minorly different whitespacing:
```
  sig {returns(Integer)}
```
Note lack of spaces in curlies. Would you be up to make your rbi's  consistent with existing ones?

**Comment by connorshea** [comment]
@DarkDimius fixed the formatting 👍

**Comment by DarkDimius** [comment]
Thank you!

**Pull request merged**